src/core/domain.py

```diff
@@ -258,3 +265,174 @@ def zookeeper_connected(self) -> bool:
     def __bool__(self) -> bool:
         """Return whether this class object has sufficient information."""
         return self.zookeeper_connected
+
+
+@dataclass
+class TLSInfo:
+    """Class representing a information related to tls."""
+
+    keystore_password: str
+    trustore_password: str
+
+
+@dataclass
+class SANs:
+    """Subject Alternative Name (SAN)s used to create multi-domains certificates."""
+
+    sans_ip: list[str]
+    sans_dns: list[str]
+
+
+class KyuubiServer(RelationState):
+    """State collection metadata for a charm unit."""
+
+    def __init__(
+        self,
+        relation: Relation | None,
+        data_interface: Data,
+        component: Unit,
+    ):
+        super().__init__(relation, data_interface, component)
+        self.unit = component
+        self.k8s = ServiceManager(
```

Comment:
**todo** This is a similar scope cripple as in the PR from @theoctober19th . I suppose we should figure out what to do in the other PR and then rebase and refactor accordingly.